### ISSUE TITLE
Fastlane failing at step Pilot with Xcode 8 

### ISSUE THREAD

**Issue by rmato** [fastlane/fastlane#6202]
### New Issue Checklist
- [ x ] Updated fastlane to the latest version
- [ x ] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
### Issue Description

Fastlane fails at step `pilot`. I'm using `match` ( with certificates and provisioning profiles stored in a private repo) + `gym` + `pilot`. Previously I had an error 65 on step `gym` with "Automatically manage signing" in Xcode 8. Now I switched to manual code signing ,set the correct provisioning profiles (debug/release) and `gym` step worked fine, but `pilot` step is failing with code signing problems. 
##### Complete output when running fastlane, including the stack trace and command used

```
[11:17:56]: -------------------
[11:17:56]: --- Step: pilot ---
[11:17:56]: -------------------
[11:17:56]: Login to iTunes Connect (ios@tril.us)
[11:18:01]: Login successful
[11:18:02]: Ready to upload new build to TestFlight (App: 1044835201)...
[11:18:02]: Going to upload updated app to iTunes Connect
[11:18:02]: This might take a few minutes. Please don't interrupt the script.
[11:19:12]: [Transporter Error Output]: ERROR ITMS-90161: "Invalid Provisioning Profile. The provisioning profile included in the bundle com.tril.Tril.qa [Payload/Tril QA.app] is invalid. [Missing code-signing certificate]. A Distribution Provisioning profile should be used when submitting apps to the App Store. For more information, visit the iOS Developer Portal."
[11:19:12]: Transporter transfer failed.
[11:19:12]: 
[11:19:12]: ERROR ITMS-90161: "Invalid Provisioning Profile. The provisioning profile included in the bundle com.tril.Tril.qa [Payload/Tril QA.app] is invalid. [Missing code-signing certificate]. A Distribution Provisioning profile should be used when submitting apps to the App Store. For more information, visit the iOS Developer Portal."
[11:19:12]: ERROR ITMS-90161: "Invalid Provisioning Profile. The provisioning profile included in the bundle com.tril.Tril.qa [Payload/Tril QA.app] is invalid. [Missing code-signing certificate]. A Distribution Provisioning profile should be used when submitting apps to the App Store. For more information, visit the iOS Developer Portal."
Return status of iTunes Transporter was 1: ERROR ITMS-90161: "Invalid Provisioning Profile. The provisioning profile included in the bundle com.tril.Tril.qa [Payload/Tril QA.app] is invalid. [Missing code-signing certificate]. A Distribution Provisioning profile should be used when submitting apps to the App Store. For more information, visit the iOS Developer Portal."
The call to the iTMSTransporter completed with a non-zero exit status: 1. This indicates a failure.
[11:19:12]: Variable Dump:
[11:19:12]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios beta", :SIGH_PROFILE_TYPE=>"app-store", :BUILD_NUMBER=>"16", :IPA_OUTPUT_PATH=>"/Users/rodrigomato/Workspace/tril-ios/Tril QA.ipa", :DSYM_OUTPUT_PATH=>"/Users/rodrigomato/Workspace/tril-ios/Tril QA.app.dSYM.zip", :XCODEBUILD_ARCHIVE=>"/Users/rodrigomato/Library/Developer/Xcode/Archives/2016-09-20/Tril QA 2016-09-20 11.13.07.xcarchive", :VERSION_NUMBER=>"1.11"}
[11:19:12]: Error uploading ipa file, for more information see above

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 3    | cocoapods                           | 10          |
| 4    | match                               | 33          |
| 5    | increment_build_number              | 1           |
| 6    | unlock_keychain                     | 0           |
| 7    | gym                                 | 142         |
| 8    | prompt                              | 150         |
| 9    | get_version_number                  | 0           |
| 10   | pilot                               | 75          |
+------+-------------------------------------+-------------+

[11:19:12]: fastlane finished with errors

```
### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

Fastfile:

``` ruby
  lane :beta do
    ENV["APP_ID"] = "com.tril.Tril.qa"

    # Make sure we don't have any uncommited changes on develop and are up to date with the remote
    #ensure_git_status_clean
    #ensure_git_branch(branch: 'develop')
    #git_pull
    #push_to_git_remote

    match(type: "appstore")

    # Increment the Build number on TestFlight
    build = increment_build_number #({ build_number: latest_testflight_build_number + 1 })

    # Build APP
    gym(scheme: "Tril QA",use _legacy_build_api: true)

    # Ask the user of fastlane to enter a changelog
    changelog = prompt(text: "Enter the change log: ", multi_line_end_keyword: "END")
    version = get_version_number(xcodeproj: "Tril.xcodeproj")
    # changelog = "New Tril QA Beta Build "

    # Upload to TestFlight
    pilot #(skip_waiting_for_build_processing: true)

    # Distribute via Crashlytics
    crashlytics(crashlytics_path: "./Frameworks/Crashlytics.framework/", 
         api_token: "---", 
         build_secret: "---",
             notifications: true,
                notes: changelog)

    # Post Slack notification
    slack(
      channel: "development",
      # default_payloads: [], # reduce the notification to the minimum
      message: "Successfully distributed Beta Build #{build} :rocket:",
      payload: {
        "Changes" => changelog
      }
    )
  end
```
### Environment

fastlane version (run `fastlane -v`): fastlane 1.103.0

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)? NO

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)? NO

**Comment by Sreilys** [comment]
Same problem.. any new suggestions?

**Comment by TKBurner** [comment]
@Sreilys Thanks for following up on this. Do you run into this issue on all projects? Does the match repo contain your certificate? Have you run `match development` on this project before? 

Thanks!

**Comment by Sreilys** [comment]
Thanks for reply back. I want to say that I am pretty new with the topic, so it could be that I do something wrong somewhere.
Is a cordova project. My match repo contains the certificate and I've nuked and match again, and the match step works fine.
The pilot step fails with the error above, yesterday I was able to upload with xcode to iTunesConnect, but I would love to do the process with fastlane, and also understand what the problem could be.
I've updated fastlane version, el capitan, and xcode 8.
